chaste_codegen/chaste_model.py

```diff
@@ -350,103 +318,151 @@ def _get_modifiable_parameters(self):
+        self._model.add_equation(sp.Eq(membrane_stimulus_current_converted, GetIntracellularAreaStimulus))
+
+        # Annotate the converted stimulus current as derived quantity
+        self._set_is_metadata(membrane_stimulus_current_converted, 'derived-quantity')
+        return membrane_stimulus_current_converted, \
+            (self._membrane_stimulus_current_orig.units, membrane_stimulus_current_converted.units)
+
     def _get_membrane_capacitance(self):
-        """ Find membrane_capacitance if the model has it"""
-        equation, capacitance_factor = None, 1.0
-        # add membrane_capacitance if the model has it
+        """ Find membrane_capacitance if the model has it and convert it to uA/uA_per_cm2 if necessary"""
         try:
-            # add membrane_capacitance factor
-            membrane_capacitance = self._model.get_variable_by_ontology_term((self._OXMETA, "membrane_capacitance"))
-            if membrane_capacitance is not None:
-                current_units = self._model.units.evaluate_units(membrane_capacitance)
-                capacitance_eqs = [eq for eq in self._model.get_equations_for([membrane_capacitance])
-                                   if eq.lhs == membrane_capacitance]
-                assert len(capacitance_eqs) == 1, 'Expecting exactly 1 defining equation expected'
-                equation = capacitance_eqs[0]
-                for desired_unit_info in [unit_dict['units'] for unit_dict in self._STIM_UNITS['membrane_capacitance']]:
-                    desired_units = self._units.get_unit(desired_unit_info)
-                    if current_units.dimensionality == desired_units.dimensionality:
-                        capacitance_factor = \
-                            self._model.units.get_conversion_factor(from_unit=current_units, to_unit=desired_units)
-                        if capacitance_factor != 1.0:
-                            warning = 'converting capacitance from ' + str(current_units) + ' to ' + str(desired_units)
-                            self._logger.info(warning)
-                        break
+            capacitance = self._model.get_variable_by_ontology_term((self._OXMETA, 'membrane_capacitance'))
+            return self._model.convert_variable(capacitance, self.uA / self.uA_per_cm2, DataDirectionFlow.OUTPUT)
         except KeyError:
-            # no membrane_capacitance
-            pass
-        return equation, capacitance_factor
+            self._logger.info(self._model.name + ' has no capacitance tagged')
+            return None
+        except DimensionalityError:
+            self._logger.info(self._model.name + ' has capacitance in incompatible units, skipping')
+            return None
 
-    def _get_stimulus(self):
-        """ Store the stimulus currents in the model"""
-        # get stimulus parameters
-        has_stim = True
-        stim_param = []
-        for tag in self._STIM_METADATA_TAGS:
+    def _get_stimulus_prams(self):
```

Comment:
```suggestion
    def _get_stimulus_params(self):
```

Reply:
Why do we need this method given the one below also returns the parameters, but converted?

Reply:
good point looks like we don't anymore

Reply:
seems maybe we don't

Reply:
seems we don't really